build/source/netcdf/def_output.f90

```diff
@@ -181,131 +198,140 @@ subroutine ini_create(nHRU,nSoil,infile,err,message)
- subroutine def_variab(infile,hruDesire,timeDesire,metadata,ivtype,err,message)
- USE data_types,only:var_info                       ! derived type for metadata
+ subroutine def_variab(ncid,iFreq,hruDesire,timeDesire,metaData,ivtype,err,message)
+ USE var_lookup,only:iLookvarType                   ! look up structure for variable typed
+ USE data_types,only:var_info                       ! derived type for metaData
+ USE var_lookup,only:iLookStat                      ! index into stats structure
+ USE var_lookup,only:maxVarStat                     ! # of available stats
+ USE get_ixName_module,only:get_varTypeName         ! to access type strings for error messages
+ USE get_ixname_module,only:get_statName            ! statistics names for variable defs in output file
  implicit none
  ! input
- character(*), intent(in)      :: infile            ! filename
- integer(i4b), intent(in)      :: hruDesire         ! variable to define if we desire the HRU dimension
- integer(i4b), intent(in)      :: timeDesire        ! variable to define if we desire the time dimension
- type(var_info),intent(in)     :: metadata(:)       ! metadata structure for a given variable
- integer(i4b),intent(in)       :: ivtype            ! variable type
+ integer(i4b)  ,intent(in)     :: ncid              ! netcdf file id
+ integer(i4b)  ,intent(in)     :: iFreq             ! frequency of current file
+ integer(i4b)  ,intent(in)     :: hruDesire         ! variable to define if we desire the HRU dimension
+ integer(i4b)  ,intent(in)     :: timeDesire        ! variable to define if we desire the time dimension
+ type(var_info),intent(inout)  :: metaData(:)       ! metaData structure for a given variable
+ integer(i4b)  ,intent(in)     :: ivtype            ! variable type
  ! output
  integer(i4b),intent(out)      :: err               ! error code
  character(*),intent(out)      :: message           ! error message
  ! local
- integer(i4b)                  :: ivar              ! variable index
+ integer(i4b)                  :: iVar              ! variable index
+ integer(i4b)                  :: iStat             ! stat index
  integer(i4b),allocatable      :: dimensionIDs(:)   ! vector of dimension IDs
- integer(i4b)                  :: ncid              ! NetCDF file ID
  integer(i4b)                  :: iVarId            ! variable ID
  character(LEN=256)            :: cmessage          ! error message of downwind routine
+ character(LEN=256)            :: catName           ! full variable name
  ! initialize error control
  err=0; message='def_variab/'
 
- ! open NetCDF file
- err = nf90_open(infile,nf90_write,ncid)
- call netcdf_err(err,message); if (err/=0) return
-
  ! allow re-definition of variables
  err = nf90_redef(ncid); call netcdf_err(err,message); if (err/=0) return
 
- ! loop through metadata
- do ivar=1,size(metadata)
+ ! loop through metaData
+ do iVar = 1,size(metaData)
```

Comment:
Please don't commit whitespace changes unless they really matter. The next person is going to get rid of the spaces again (since that's how it's done in much of the code) and it just clutters the commits and reviews.